google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/MessageDispatcher.java

```diff
@@ -409,34 +474,45 @@ public void run() {
   /** Compute the ideal deadline, set subsequent modacks to this deadline, and return it. */
   @InternalApi
   int computeDeadlineSeconds() {
-    int sec = ackLatencyDistribution.getPercentile(PERCENTILE_FOR_ACK_DEADLINE_UPDATES);
-
-    if ((maxSecondsPerAckExtension > 0) && (sec > maxSecondsPerAckExtension)) {
-      sec = maxSecondsPerAckExtension;
+    int deadlineSeconds = ackLatencyDistribution.getPercentile(PERCENTILE_FOR_ACK_DEADLINE_UPDATES);
+
+    // Bound deadlineSeconds by extensions
+    if (!maxDurationPerAckExtensionDefaultUsed
+        && (deadlineSeconds > maxDurationPerAckExtensionSeconds)) {
+      deadlineSeconds = maxDurationPerAckExtensionSeconds;
+    } else if (deadlineSeconds < minDurationPerAckExtensionSeconds) {
```

Comment:
It does not actually matter at this point. The check for `maxDurationPerAckExtensionDefaultUsed` is done because the default max is 0 which would make the bounds behave incorrectly.